src/main/kotlin/org/malachite/estella/services/TaskService.kt

```diff
@@ -98,38 +142,49 @@ class TaskService(
             .let {
                 updateTask(
                     taskId,
-                    it.copy(tests = SerialBlob(Base64.getEncoder().encode(tests.toString().encodeToByteArray())))
+                    it.copy(tests = tests.encodeToJson().toByteArray().toTypedArray())
                 )
             }
 
     fun addResult(result: TaskResultRabbitDTO) {
         val taskStage = taskStageService.getTaskStage(result.solverId)
         val task = getTaskById(result.taskId).toTask()
         val startTime = result.startTime
-                ?.takeIf { it != "null" }
-                ?.let { Timestamp.valueOf(it) }
+            ?.takeIf { it != "null" }
+            ?.let { Timestamp.valueOf(it) }
         val endTime = result.endTime
-                ?.takeIf { it != "null" }
-                ?.let { Timestamp.valueOf(it) }
+            ?.takeIf { it != "null" }
+            ?.let { Timestamp.valueOf(it) }
 
         taskStageService.addResult(TaskResult(null,
-                SerialBlob(Base64.getEncoder().encode(result.results.toByteArray())),
-                SerialClob(result.code.toCharArray()),
-                startTime,
-                endTime,
-                task,
-                taskStage
+            SerialBlob(Base64.getEncoder().encode(result.results.toByteArray())),
+            SerialClob(result.code.toCharArray()),
+            startTime,
+            endTime,
+            task,
+            taskStage
         ))
     }
 
 
     private fun checkTestsFormat(tests: String) = try {
         val decoded = String(Base64.getDecoder().decode(tests))
-        Json.decodeFromString<List<TaskTestCaseDto>>(decoded)
-        true
+        TaskTestCaseDto.decodeFromJson(decoded)
     } catch (e: SerializationException) {
         e.printStackTrace()
-        false
+        throw InvalidTestFileException()
     }
 
+    fun checkOrganizationRights(organizationUuid: UUID, taskId: Int): TaskService =
```

Comment:
To zwraca TaskService? To mi wygląda bardziej na buildera, ale jeśli tego potrzebujesz, to ok

Reply:
To taki assert w stylu `checkDevPassword`